crates/holochain/src/core/workflow/sys_validation_workflow/types.rs

```diff
@@ -9,6 +9,6 @@ pub(crate) enum Outcome {
     /// be found currently on the DHT.
     /// Note this is not proof it doesn't exist.
     MissingDhtDep(AnyDhtHash),
-    /// Moves to integration with status rejected
-    Rejected,
+    /// Moves to integration with status rejected, with an informational reason
+    Rejected(String),
```

Comment:
This allows me to greatly improve the tests. There are many reasons why an op might be rejected and asserting the actual reason is much more reliable than hoping everything is correct except the one reason you meant to test for. 

This did catch an issue with one of the tests, so a worthwhile change I think even though it's never used in the workflow code.

Reply:
How about going one step further and using the `ValidationOutcome` itself in this `Rejected` variant?

Reply:
That would be better and I did start there but the ValidationOutcome embeds several types from other modules that aren't `Clone` which the errors needs to be in the current code. I think the reasonable change would be to reduce how much gets put in the error message because I'm not sure how informative entire structs are but it's a more involved change than  I wanted to get into for these tests. What do you think?

Reply:
Sounds good to me, we can refactor this if needed.